### PR TITLE
Add the missing test for ERC721Holder

### PR DESCRIPTION
**Pull request by elopio**
This is one step towards #1147 

# 🚀 Description

Add test coverage for ERC721Holder.

- [x] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [x] ✅ I've added tests where applicable to test my new functionality.
- [x] 📖 I've made sure that my contracts are well-documented.
- [x] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: test/token/ERC721/ERC721Holder.behavior.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const ERC721Holder = artifacts.require('ERC721Holder.sol');
+
+require('chai')
+  .should();
+
+function shouldBehaveLikeERC721Holder (accounts) {
+  const tokenId = 1;
+  const creator = accounts[0];
```
Comment (nventuro) [inline test/token/ERC721/ERC721Holder.behavior.js]: :scream: 

Consider destructuring the `accounts` array (e.g. change the function declaration to  `function shouldBehaveLikeERC721Holder ([creator])`), see #1137 for reference :sweat_smile:

File: test/token/ERC721/ERC721Holder.behavior.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const ERC721Holder = artifacts.require('ERC721Holder.sol');
+
+require('chai')
+  .should();
+
+function shouldBehaveLikeERC721Holder (accounts) {
```
Comment (nventuro) [inline test/token/ERC721/ERC721Holder.behavior.js]: I'm not sure this name is accurate: the token isn't behaving like an ERC721 Holder, it is instead interacting with one. 

I think we should have a test for the receiver itself, that calls the function and checks the return value, and one for the token, checking that `safeTransferFrom` works properly (i.e. rejects an invalid return value, accepts the correct one). @shrugs is the local ERC721 expert though, so I'll defer to his wisdom.

Comment (elopio) [reply test/token/ERC721/ERC721Holder.behavior.js]: maybe I can rename it to: shouldWorkWithERC721Holder.

Comment (nventuro) [reply test/token/ERC721/ERC721Holder.behavior.js]: I checked the tests in some more detail: we're already testing that `safeTransferFrom` in the `describe('via safeTransferFrom')` block, what we need to test is that the ERC721Holder works with a compliant ERC721 token (i.e. it returns the correct value). Does that make sense?

Comment (elopio) [reply test/token/ERC721/ERC721Holder.behavior.js]: It does. But I think that the best way to test that the correct value is returned from the Holder contract is to check that safeTransferFrom succeeds. Just calling onReceived directly and verifying that a magic value is returned is too low level, IMO.

Comment (nventuro) [reply test/token/ERC721/ERC721Holder.behavior.js]: My point was that it's not a token test, but a holder test - it belongs in a separate test file :stuck_out_tongue:

Comment (elopio) [reply test/token/ERC721/ERC721Holder.behavior.js]: Yes, that sounds right. I am just wondering if I should keep this style of tests/behavior, even if it's only one simple test.

**Comment by elopio** [comment]
@nventuro one more time, please.

**Pull request merged**